expando/node_padding.c

```diff
@@ -79,18 +79,21 @@ int pad_string(const struct ExpandoNode *node, struct Buffer *buf, int max_cols)
   const int pad_cols = mutt_strnwidth(node->start, pad_len);
   int total_cols = 0;
 
-  while (pad_cols <= max_cols)
+  if (pad_len != 0)
   {
-    buf_addstr_n(buf, node->start, pad_len);
+    while (pad_cols <= max_cols)
+    {
+      buf_addstr_n(buf, node->start, pad_len);
 
-    max_cols -= pad_cols;
-    total_cols += pad_cols;
+      max_cols -= pad_cols;
+      total_cols += pad_cols;
+    }
   }
 
-  for (; (max_cols > 0); buf++, max_cols--)
+  if (max_cols > 0)
   {
-    buf_addch(buf, ' ');
-    total_cols++;
+    buf_add_printf(buf, "%*s", max_cols, "");
```

Comment:
I think the padding with spaces was @flatcap s addition and padding with "empty string" is a bit stange (for me), why dont we use `buf_add_printf(buf, "%*s", max_cols, " ");`  instead?

Reply:
I think you are misunderstanding the format string. The "padding with spaces" comes from the fact that the minimum field width will add spaces if the provided string doesn't fill the width. It doesn't repeat the provided string to fill in the width. So, to me, providing an empty string is less confusing than printing one space then filling in the remaining width with spaces, because the two uses of a space aren't related:
* don't print anything, but fill in with spaces (because that's what the width specifier does)
* print a space, then fill in with spaces (not related to the fact that I printed a space anyway)

not my clearer answer ever, I realize.. I hope it clarifies anyway

Reply:
Sure, I understood your intent. It it perfectly fine solution, I just wanted to know the reasoning behind it :)